azure-pipelines.yml

```diff
@@ -103,3 +105,8 @@ jobs:
           --configuration $(_BuildConfig)
           --prepareMachine
         displayName: Unix Build / Publish
+# Define all validation legs here in template form
+- ${{ if and(ne(variables['System.TeamProject'], 'public'), notin(variables['Build.Reason'], 'PullRequest')) }}:
```

Comment:
For now, yes, I tried https://github.com/dotnet/arcade/blob/master/Documentation/AzureDevOps/SendingJobsToHelix.md#internal-builds and did not work for me, so in order for me to keep moving forward I used Anonymous access

Reply:
That doesn't make sense.  That variable group shouldn't even be available in the internal project

Reply:
cc/ @MattGal 

Matt Gal is seeing some other issues w/ permissions as well.  We do need to get this sorted....

Reply:
I see the "Helix Anonymous" variable group is present in the internal project.  But it shouldn't be there.  someone just added it.

Reply:
I wonder if it was done as some sort of workaround for when Winforms started working on adding Helix while their repo was private and their CI ran in the internal project?

Reply:
Ok, so I guess the clean thing to do is remove Anonymous from internal and figure out why https://github.com/dotnet/arcade/blob/master/Documentation/AzureDevOps/SendingJobsToHelix.md#internal-builds is not working. Will do that in this PR

Reply:
Not to point fingers, but the variable group says it was last modified by some Juan dude yesterday.  :) That's actually a positive if he added it because it means we can just delete it and not worry about anyone else having taken a dependency on it that we could break.  If it has been there longer, then that's a problem.

Reply:
I modified the secret in there trying to test the internal scenario which failed, if remember correctly it was there already(?) Don't really recall adding it but hell I don't care I can take the blame if this will unblock progress on this

Reply:
" Helix operation returned 'Unauthorized'. Did you forget to set HelixAccessToken?" -> https://dnceng.visualstudio.com/internal/_build/results?buildId=58918

Reply:
The key is named "HelixApiAccessToken", did you set that and pass it?

Reply:
Well according to https://github.com/dotnet/arcade/blob/master/Documentation/AzureDevOps/SendingJobsToHelix.md#internal-builds " $HelixAccessToken is automatically injected into the environment" but I'll set that then